DatadogCore/Tests/Datadog/LoggerTests.swift

```diff
@@ -875,6 +876,45 @@ class LoggerTests: XCTestCase {
         logMatchers[1].assertNoValue(forKey: "dd.span_id")
     }
 
+    func testGivenBundlingWithTraceEnabledAndOpenTelemetryTracerRegistered_whenSendingLog_itContainsActiveSpanAttributes() throws {
+        core.context = .mockAny()
+
+        Logs.enable(in: core)
+        Trace.enable(in: core)
+
+        // given
+        let logger = Logger.create(in: core)
+        OpenTelemetry.registerTracerProvider(
+            tracerProvider: OTelTracerProvider(in: core)
+        )
+
+        let tracer = OpenTelemetry
+            .instance
+            .tracerProvider
+            .get(instrumentationName: "", instrumentationVersion: nil)
+
+        // when
+        let span = tracer.spanBuilder(spanName: "span")
+            .setActive(true)
+            .startSpan()
+        logger.info("info message 1")
+        span.end()
+        logger.info("info message 2")
```

Comment:
food for thought/ One thing that bothers me is that in V2 the "active span" [we store in `DatadogContext`](https://github.com/DataDog/dd-sdk-ios/blob/eb5515c6bf28f3cd6faa15719a970f39c46183d0/DatadogTrace/Sources/DatadogTracer.swift#L134) is actually the "last span marked active in `DatadogTrace`" not the "active span in current thread". To illustrate this, following change will keep the test green although it shouldn't if we stick to the definition of "active span" per OT or OTel.

Not saying there's something to do in this PR, just bringing this problem up as we're stepping over it. I wonder what's your take on this @ganeshnj.

```swift
let queue = DispatchQueue(label: "foo")
var span: Span!

queue.sync {
    span = tracer.spanBuilder(spanName: "span")
        .setActive(true)
        .startSpan()
}
logger.info("info message 1")
queue.sync {
    span.end()
}
logger.info("info message 2")
```

Reply:
agree, current version is very naive. This will be more troublesome with swift-concurrency (actors specifically).

I have this back of my mind, and hopefully will follow up once otel topic is out.